janusgraph-berkeleyje/src/main/java/org/janusgraph/diskstorage/berkeleyje/BerkeleyJEKeyValueStore.java

```diff
@@ -191,21 +197,31 @@ public void remove() {
     }
 
     @Override
-    public void insert(StaticBuffer key, StaticBuffer value, StoreTransaction txh) throws BackendException {
-        insert(key, value, txh, true);
+    public void insert(StaticBuffer key, StaticBuffer value, StoreTransaction txh, Integer ttl) throws BackendException {
+        insert(key, value, txh, true, ttl);
     }
 
-    public void insert(StaticBuffer key, StaticBuffer value, StoreTransaction txh, boolean allowOverwrite) throws BackendException {
+    public void insert(StaticBuffer key, StaticBuffer value, StoreTransaction txh, boolean allowOverwrite, Integer ttl) throws BackendException {
         Transaction tx = getTransaction(txh);
         try {
             OperationStatus status;
 
             log.trace("db={}, op=insert, tx={}", name, txh);
 
-            if (allowOverwrite)
-                status = db.put(tx, key.as(ENTRY_FACTORY), value.as(ENTRY_FACTORY));
-            else
-                status = db.putNoOverwrite(tx, key.as(ENTRY_FACTORY), value.as(ENTRY_FACTORY));
+            WriteOptions writeOptions = null;
+
+            if (ttl != null && ttl > 0) {
+                int hoursTtl = convertSecondsToHoursTtl(ttl);
+                writeOptions = new WriteOptions().setTTL(hoursTtl, TimeUnit.HOURS);
+            }
+            if (allowOverwrite) {
+                OperationResult result = db.put(tx, key.as(ENTRY_FACTORY), value.as(ENTRY_FACTORY), Put.OVERWRITE, writeOptions);
+                EnvironmentFailureException.assertState(result != null);
```

Comment:
This assert is used here to get a `DatabaseException` in case the write was not successful to catch it and then rethrow it as a `PermanentBackendException` at the end of the method, right?

Could you please explain in general why it's not sufficient any more to just use the returned status for the check below?

Reply:
I removed extra exception handling